tinygrad/runtime/ops_cpu.py

```diff
@@ -16,6 +16,10 @@ def shape_to_axis(old_shape:Tuple[int, ...], new_shape:Tuple[int, ...]) -> Tuple
   MovementOps.RESHAPE: lambda x, arg: x.reshape(arg), MovementOps.SHRINK: lambda x, arg: x[tuple(slice(p[0], p[1], None) for p in arg)],
 }
 
+def promote_types(x, y):
+  ret = np.promote_types(x.dtype,y.dtype)
+  return ret if ret != np.float64 else np.float32
```

Comment:
I think this is fine, matching torch behaviour should be fine.

Could do:
`def promote_types(x, y): (ret := np.promote_types(x.dtype, y.dtype)) if ret != np.float64 else np.float32`